samples/server/go-server/api/server/storage/project_memstore.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+
+	"github.com/grafeas/grafeas/samples/server/go-server/api/server/name"
+	"github.com/grafeas/grafeas/server-go"
+	pb "github.com/grafeas/grafeas/v1alpha1/proto"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// projectMemStore is an in-memory storage solution for Grafeas
+type projectMemStore struct {
+	projects map[string]bool
+}
+
+// NewMemStore creates a memStore with all maps initialized.
+func NewProjectMemStore() server.ProjectStorager {
+	return &projectMemStore{make(map[string]bool)}
+}
+
+// CreateProject adds the specified project to the mem store
+func (m *projectMemStore) CreateProject(pID string) error {
+	if _, ok := m.projects[pID]; ok {
+		return status.Error(codes.InvalidArgument, fmt.Sprintf("Project with name %q already exists", pID))
```

Comment:
`codes.AlreadyExists`. Use `codes.InvalidArgument` for data that is improperly formatted or makes no sense.